### PR TITLE
Modify Rule S5883(C#): Education Framework (APPSEC-61)

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
Specification ticket: [APPSEC-61](https://sonarsource.atlassian.net/browse/APPSEC-61).

### CODE REVIEW

File: rules/S5883/csharp/how-to-fix-it/dotnet.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+=== How does this work?
+
+include::../../common/fix/introduction.adoc[]
+
+:sanitizationLib: ArgumentList.Add()
+include::../../common/fix/sanitize-meta-characters.adoc[]
+
+Here `ArgumentList` takes care of escaping the passed arguments and internally
+creates a single string given to the operating system when `Process.Start()` is
```
Comment (pierre-loup-tristant-sonarsource) [inline rules/S5883/csharp/how-to-fix-it/dotnet.adoc:62]:
```suggestion
creates a single string given to the operating system when `System.Diagnostics.Process.Start()` is
```

**Review by pierre-loup-tristant-sonarsource — CHANGES_REQUESTED** [review]
Please fix:
- Broken rendering after the payload
- Payload itself
- Typo in the parameter name in the compliant code examples

Other comment are minor.

File: rules/S5883/csharp/how-to-fix-it/dotnet.adoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+It is vulnerable to argument injection because user-controlled data is inserted
+directly into the arguments of a process call without sanitization. +
+The application assumes that incoming data always consists of a specific range
+of characters and ignores that some characters might force the `find` command
+to start a shell.
+
+In this particular case, an attacker may exploit the query with the following string:
+
+-----
+-exec /bin/sh \; -quit
```
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S5883/csharp/how-to-fix-it/dotnet.adoc:15]: ah yeah I like the rm one, it's clearer

File: rules/S5883/common/fix/sanitize-meta-characters.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
+==== Neutralize special characters
+
+When the application needs to execute complex commands that require
+user-controlled data to be inserted directly into a process invocation command,
+it is cleanest to use process call libraries that provide argument sanitization
+internally, such as {sanitizationLib}.
```
Comment (loris-s-sonarsource) [inline rules/S5883/common/fix/sanitize-meta-characters.adoc:6]:
```suggestion
internally, such as `{sanitizationLib}`.
```

File: rules/S5883/csharp/how-to-fix-it/dotnet.adoc [hunk 4]
```diff
@@ -3,16 +3,16 @@
...
 name of a file to find on the system.
 
-It is vulnerable to argument injection because user-controlled data is inserted
+It is vulnerable to arguments injection because untrusted data is inserted
 directly into the arguments of a process call without sanitization. +
 The application assumes that incoming data always consists of a specific range
 of characters and ignores that some characters might force the `find` command
 to start a shell.
 
 In this particular case, an attacker may exploit the query with the following string:
```
Comment (pierre-loup-tristant-sonarsource) [inline rules/S5883/csharp/how-to-fix-it/dotnet.adoc:12]: Maybe you can mention that the payload will remove all file for the application root directory?

**Review by pierre-loup-tristant-sonarsource — APPROVED** [review]
LGTM, one minor comment.

File: rules/S5883/csharp/how-to-fix-it/dotnet.adoc (hunk 4 above)
Comment (loris-s-sonarsource) [reply rules/S5883/csharp/how-to-fix-it/dotnet.adoc:12]: yup, my bad !

File: rules/S5883/csharp/how-to-fix-it/dotnet.adoc [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+It is vulnerable to arguments injection because untrusted data is inserted
+directly into the arguments of a process call without sanitization. +
+The application assumes that incoming data always consists of a specific range
+of characters and ignores that some characters might force the `find` command
+to start a shell.
+
+In this particular case, an attacker may exploit the query with the following string:
+
+----
+* -exec rm {} ;
```
Comment (loris-s-sonarsource) [inline rules/S5883/csharp/how-to-fix-it/dotnet.adoc:15]:
```suggestion
'*' -exec echo rm -rf {} \; 
```

File: rules/S5883/csharp/how-to-fix-it/dotnet.adoc [hunk 6]
```diff
@@ -0,0 +1,65 @@
+=== How to fix it in .NET
+
+The following code uses the `find` command and expects the user to enter the
+name of a file to find on the system.
+
+It is vulnerable to arguments injection because untrusted data is inserted
+directly into the arguments of a process call without sanitization. +
+The application assumes that incoming data always consists of a specific range
+of characters and ignores that some characters might force the `find` command
+to start a shell.
```
Comment (loris-s-sonarsource) [inline rules/S5883/csharp/how-to-fix-it/dotnet.adoc:10]:
```suggestion
to forcefully remove files.
```

File: rules/S5883/csharp/how-to-fix-it/dotnet.adoc [hunk 7]
```diff
@@ -0,0 +1,65 @@
+=== How to fix it in .NET
+
+The following code uses the `find` command and expects the user to enter the
+name of a file to find on the system.
+
+It is vulnerable to arguments injection because untrusted data is inserted
+directly into the arguments of a process call without sanitization. +
+The application assumes that incoming data always consists of a specific range
+of characters and ignores that some characters might force the `find` command
+to forcefully remove files in `/some/folder`.
```
Comment (loris-s-sonarsource) [inline rules/S5883/csharp/how-to-fix-it/dotnet.adoc:10]:
```suggestion
to remove files in `/some/folder`.
```

File: rules/S5883/csharp/how-to-fix-it/dotnet.adoc [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+The following code uses the `find` command and expects the user to enter the
+name of a file to find on the system.
+
+It is vulnerable to arguments injection because untrusted data is inserted
+directly into the arguments of a process call without sanitization. +
+The application assumes that incoming data always consists of a specific range
+of characters and ignores that some characters might force the `find` command
+to remove files in `/some/folder`.
+
+In this particular case, an attacker may exploit the query with the following string:
```
Comment (loris-s-sonarsource) [inline rules/S5883/csharp/how-to-fix-it/dotnet.adoc:12]:
```suggestion
of characters and ignores that some characters might force the `find` command
to start a shell.

In this particular case, an attacker may remove files in `/some/folder` with the following string:
```

**Pull request merged**